easybuild/tools/modules.py

```diff
@@ -1433,30 +1457,12 @@ def use(self, path, priority=None):
             if os.environ.get('__LMOD_Priority_MODULEPATH'):
```

Comment:
We're using `__LMOD_Priority_MODULEPATH` in multiple places, so this should become a class constant in the `Lmod` class imho, and maybe refer to https://github.com/TACC/Lmod/issues/509 to clarify that it's OK to rely on this?

Reply:
I refactored this into `_has_module_paths_with_priority` so make it even more clear and provide a single point of use